src/services/accounts/AccountsBalanceInfoService.ts

```diff
@@ -161,14 +174,89 @@ export class AccountsBalanceInfoService extends AbstractService {
 				at,
 				nonce,
 				tokenSymbol: token,
-				free,
-				reserved,
-				miscFrozen,
-				feeFrozen,
-				locks,
+				free: this.inDenominationBal(denominate, free, decimal),
+				reserved: this.inDenominationBal(denominate, reserved, decimal),
+				miscFrozen: this.inDenominationBal(denominate, miscFrozen, decimal),
+				feeFrozen: this.inDenominationBal(denominate, feeFrozen, decimal),
+				locks: this.inDenominationLocks(denominate, locks, decimal),
 			};
 		} else {
 			throw new BadRequest('Account not found');
 		}
 	}
+
+	/**
+	 * Apply a denomination to a balance depending on the chains decimal value.
+	 *
+	 * @param balance free balance available encoded as Balance
+	 * @param dec The chains given decimal value
```

Comment:
If you could explain that `dec` is decimal tokens in (ss58 context) which is something that applied to `base unit` then I'm good with this.

AFAIU the `decimal tokens` must > 0

Reply:
added